pkg/policy/l4.go

```diff
@@ -165,13 +178,17 @@ type PerSelectorPolicy struct {
 	IsDeny bool `json:",omitempty"`
 }
 
-// Equal returns true if 'a' and 'b' represent the same L7 Rules
-func (a *PerSelectorPolicy) Equal(b *PerSelectorPolicy) bool {
-	return a == nil && b == nil || a != nil && b != nil &&
-		a.TerminatingTLS.Equal(b.TerminatingTLS) &&
-		a.OriginatingTLS.Equal(b.OriginatingTLS) &&
-		a.IsDeny == b.IsDeny &&
-		reflect.DeepEqual(a.L7Rules, b.L7Rules)
+// Equal is used in tests for the check.Checker.
+func (in *PerSelectorPolicy) Equal(o *PerSelectorPolicy) bool {
+	return in.DeepEqual(o)
```

Comment:
This is not handling the case `in == nil` like the old `Equal()` did. If this would, then this becomes the same as `DeepEqual()` below.

Reply:
I don't understand the comment, can you clarify it? This `Equal` needs to be present because of `check.Checker`.

Reply:
The old implementation allowed `in` to be `nil`, the new does not.

Reply:
Not sure it is needed though, if you did not see any panics due to `nil` pointer dereference in `in.DeepEqual()`?